ktlint/src/main/kotlin/com/pinterest/ktlint/internal/KtlintCommandLine.kt

```diff
@@ -291,13 +291,6 @@ internal class KtlintCommandLine {
         val stdinPatterns: Set<String> = readPatternsFromStdin()
         patterns.addAll(stdinPatterns)
 
-        // Set default value to patterns only after the logger has been configured to avoid a warning about initializing
-        // the logger multiple times
-        if (patterns.isEmpty()) {
-            logger.info { "Enable default patterns $DEFAULT_PATTERNS" }
-            patterns = ArrayList(DEFAULT_PATTERNS)
-        }
```

Comment:
What about something like:
```
        // Set default value to patterns only after the logger has been configured to avoid a warning about initializing
        // the logger multiple times
       patterns = when {
           stdinPatterns.isNotEmpty() ->
                // TBD: log warning in case patterns were also specified at command line?
                stdinPatterns
           patterns.isEmpty() -> {
                logger.info { "Enable default patterns $DEFAULT_PATTERNS" }
                ArrayList(DEFAULT_PATTERNS)
           }
           else -> {
               // Keep patterns specified at command line
              patterns
           }
       }
```